native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java

```diff
@@ -266,7 +266,12 @@ public static Object copy(BString sourcePath, BString destinationPath, BString..
                 Files.walkFileTree(srcPath, new RecursiveFileCopyVisitor(srcPath, destPath, ops));
             }
         } catch (IOException ex) {
-            return FileUtils.getBallerinaError(FileConstants.FILE_SYSTEM_ERROR, ex);
+            if (ex.toString().contains("NoSuchFileException")) {
+                return FileUtils.getBallerinaError(FileConstants.FILE_NOT_FOUND_ERROR,
+                        "The target directory does not exist: " + ex.getMessage());
+            }
+            return FileUtils.getBallerinaError(FileConstants.FILE_SYSTEM_ERROR,
+                    "An error occurred when copying the file/s: " + ex.getMessage());
```

Comment:
} catch (NoSuchFileException ex) {
        return FileUtils.getBallerinaError(FileConstants.FILE_NOT_FOUND_ERROR,
                        "The target directory does not exist: " + ex.getMessage());
} catch (IOException ex) {
        return FileUtils.getBallerinaError(FileConstants.FILE_SYSTEM_ERROR,
                    "An error occurred when copying the file/s: " + ex.getMessage());
}

Reply:
Shall we do like this

Reply:
Updated.